src/screens/TokenDetail.js

```diff
@@ -0,0 +1,685 @@
+import ModalConfirm from '../components/ModalConfirm';
+import ModalEditToken from '../components/ModalEditToken';
+import ModalPin from '../components/ModalPin';
+import QRCode from 'qrcode.react';
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+import HathorAlert from '../components/HathorAlert';
+import { connect } from "react-redux";
+import ReactLoading from 'react-loading';
+import hathorLib from '@hathor/wallet-lib';
+
+const mapStateToProps = (state) => {
+  return {
+    historyTransactions: state.historyTransactions,
+  };
+};
+
+
+/**
+ * Screen to manage a token. Mint, melt, edit, unregister, configuration string.
+ *
+ * @memberof Screens
+ */
+class TokenDetail extends React.Component {
+  constructor(props) {
+    super(props);
+
+    /**
+     * token {Object} selected token data
+     * mintOutputs {Object} array with outputs available to mint
+     * meltOutputs {Object} array with outputs available to melt
+     * walletAmount {number} amount available of this token on this wallet
+     * destroyQuantity {number} holds the destroy quantity typed in the input
+     * action {string} selected action (mint, melt, delegate-mint, delegate-melt, destroy-mint, destroy-melt)
+     * successMessage {string} success message to show
+     * errorMessage {string} error message to show
+     * loading {boolean} if should show loading spinner
+     * pin {string} pin typed on input
+     * formValidated {boolean} if form was already validated
+     */
+    this.state = {
+      token: null,
+      mintOutputs: [],
+      meltOutputs: [],
+      walletAmount: 0,
+      destroyQuantity: 0, // this is the only one I need as state because I use this value to show message on the confirm modal
+      action: '',
+      successMessage: '',
+      errorMessage: '',
+      loading: false,
+      pin: '',
+      formValidated: false,
+    };
+
+    // Mint/melt refs
+    this.amount = React.createRef();
+    this.createAnother = React.createRef();
+    this.chooseAddress = React.createRef(); // Only for mint
+    this.address = React.createRef(); // Only for mint
+    this.addressWrapper = React.createRef(); // Only for mint
+
+    // Delegate refs
+    this.delegateAddress = React.createRef();
+    this.delegateCreateAnother = React.createRef();
+
+    // Destroy quantity ref
+    this.destroyQuantity = React.createRef();
+
+    // Form ref
+    this.form = React.createRef();
+  }
+
+  componentDidMount() {
+    const { match: { params } } = this.props;
+
+    const allTokens = hathorLib.tokens.getTokens();
+    const token = allTokens.find((data) => data.uid === params.tokenUID);
+
+    this.setState({ token }, () => {
+      this.updateTokenData();
+    });
+  }
+
+  componentDidUpdate = (prevProps) => {
+    if (this.props.historyTransactions !== prevProps.historyTransactions) {
+      this.updateTokenData();
+    }
+  }
+
+  /**
+   * Update token state after didmount or props update
+   */
+  updateTokenData = () => {
```

Comment:
This approach seems to fail when another wallet spends your authority utxo because the list will be outdated. Am I missing something?

Reply:
How would another wallet spend your authority output? You mean yourself in another device? Like the mobile app?

And even if this happens, it would update the `historyTransactions` in redux, that would call `componentDidUpdate` with new `props` and then `updateTokenData ` will be called to update the state in the screen.

Is this the problem? I am not sure I fully understand your concern

Reply:
I meant that multiple devices may have access to the same wallet (like when we are sharing the words in our testnet). But your flow seems correct. So, the screen will be updated.